.github/workflows/test.yml

```diff
@@ -24,15 +24,33 @@ jobs:
           black --check urlchecker
 
 
-  testing:
+  type_checking:
     needs: formatting
     runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Setup mypy environment
+        run: conda create --quiet --name type_checking mypy
+
+      - name: Check types with mypy
+        run: |
+          export PATH="/usr/share/miniconda/bin:$PATH"
+          source activate type_checking
+          pip install mypy
+          pip install types-requests
+          mypy urlchecker
```

Comment:
This is perfecto! It also is nice to see what the action is doing. I'm all for using external actions, but using one usually means I want to review it in detail (for security reasons) and then hard code the commit I reviewed. This is much nicer because I can see it explicitly (and could reproduce running it locally).